### DISCUSSION TITLE
Correct setting for time in germany

### DISCUSSION THREAD

**Discussion by basementmedia2** [bxparks/AceTime#63] [General]
Hi, 

first of all: great library.

I want to get the correcttime displayed for germany.
Is this the correct setting:

```
auto myTz = TimeZone::forZoneInfo(&zonedb::kZoneEurope_Berlin,
      &pacificProcessor);
    auto myTime = ZonedDateTime::forEpochSeconds(now, myTz);
```

Or is "pacificProcessor" and / or "ZoneEurope_Berlin" wrong?

Where can i find an overview of the correct timezone-wordings?

And last questions:
Once the time is correctly set and the esp will not be switched off will AceTime automatically recoginze change to summer / wintertime?

Best wishes
Daniel

**Comment by bxparks** [comment]
Yes, that looks right. I would rename `pacificProcessor` to `berlinProcessor`. It is called `pacificProcessor` in my sample code because I live in Pacific Standard/Daylight Time. So your code should look something like this:

```C++
static BasicZoneProcessor berlinProcessor;
...
void doSomethingWithTime() {
  auto berlinTz = TimeZone::forZoneInfo(
    &zonedb::kZoneEurope_Berlin, &berlinProcessor);
  auto berlinTime = ZonedDateTime::forEpochSeconds(now, berlinTz);
  [...]
}
```
(Edit: Fix typo above)

The full list of all timezones supported by `BasicZoneProcessor` is given in the header file: 
https://github.com/bxparks/AceTime/blob/develop/src/ace_time/zonedb/zone_infos.h

The full list of all timezones supported by `ExtendedZoneProcessor` is given in this header file:
https://github.com/bxparks/AceTime/blob/develop/src/ace_time/zonedbx/zone_infos.h

Yes, your ZonedDataTime will calculate summer/winter time automatically. Good luck with your project!
